### PR TITLE
Add tracked courses networking

### PR DESCRIPTION
**Pull request by haiyingweng**
## Overview

Added networking for getting currently tracked courses

## Changes Made

I hardcoded in my auth header (not included in this PR) then added request to get currently tracked courses

## Test Coverage

Manual. 

## Screenshots

<details>

  <summary>Screen Shot</summary>

  
Screen Shot 2021-04-26 at 1 12 32 AM

</details>

### CODE REVIEW

File: src/utils/requests.ts [hunk 1]
```diff
@@ -43,7 +43,10 @@ export const initializeSession = async () => { }
 
 export const updateSession = async () => { }
 
-export const getAllTrackedSections = async () => { }
+export const getAllTrackedSections = async () => {
+    const data = await get(`/users/tracking/`)
+    return data
+}
```
Comment (danielVebman) [inline src/utils/requests.ts:49]:
```suggestion
export const getAllTrackedSections = async () => await get(`/users/tracking/`)
```

Inline

File: src/types.ts [hunk 2]
```diff
@@ -17,6 +17,10 @@ export interface Section {
     title: string
 }
 
+export interface Sections {
+    sections: Section[]
+}
```
Comment (danielVebman) [inline src/types.ts:22]: What's the point of this interface? (See my other comment in `TrackedCoursesView`)

File: src/components/TrackedCourses/TrackedCoursesView.tsx [hunk 3]
```diff
@@ -2,52 +2,39 @@ import React from 'react'
...
+    awaitingSections: []
+  }
+
+  componentDidMount() {
+    this.getTrackedSections()
+  }
+
+  getTrackedSections = async () => {
+    try {
+      const data: Sections = await getAllTrackedSections()
```
Comment (danielVebman) [inline src/components/TrackedCourses/TrackedCoursesView.tsx:28]: What's the point of `Sections`? It seems like clutter. I think you should be able to say this without explicitly defining the interface:

```
const data = await getAllTrackedSections()
const available = data.sections. ...
```

**Review by danielVebman — APPROVED** [review]
Hella clean 🤩🤩🤩 🚀🔔

**Review by mathewscullin — APPROVED** [review]
Wonderful stuff 💯 I second Daniel's comments about the Sections interface, otherwise looks good!

File: src/components/TrackedCourses/TrackedCoursesView.tsx (hunk 3 above)
Comment (haiyingweng) [reply src/components/TrackedCourses/TrackedCoursesView.tsx:28]: tried to do it this way, but it doesn't seem to work with the filter function. When I filter the sections, without explicitly declaring the type, each section has type any and it throws an error for that.

**Pull request merged**